.github/workflows/benchmark.yaml

```diff
@@ -0,0 +1,28 @@
+name: Benchmark
+on:
+  push:
+    branches:
+      - main
```

Comment:
Should we run this on all branches, so that any regressions can be identified on the PRs that introduce them?

Reply:
I didn't want to slow down our PR checks too much since I really like the fast feedback loop we have at the moment. Also, we don't really have a way to actually detect a regression at the moment. 

I was thinking for now this would be something we detect when preparing a release since that's the point we'll be updating the docs with the most recent benchmarks.

Reply:
What we *could* do is pick some upper limit for each benchmark based on the results we see in CI and fail the benchmark if it goes above that limit? Say 20% above?

We could also test against a smaller row count to keep CI times down.